Analysis/src/ConstraintGraphBuilder.cpp

```diff
@@ -1067,20 +1067,38 @@ ControlFlow ConstraintGraphBuilder::visit(const ScopePtr& scope, AstStatIf* ifSt
     ScopePtr elseScope = childScope(ifStatement->elsebody ? ifStatement->elsebody : ifStatement, scope);
     applyRefinements(elseScope, ifStatement->elseLocation.value_or(ifStatement->condition->location), refinementArena.negation(refinement));
 
+    const ControlFlow guardClauseFlows = FFlag::LuauLoopControlFlowAnalysis ? ExitingControlFlows : ControlFlow::Returns | ControlFlow::Throws;
+
     ControlFlow thencf = visit(thenScope, ifStatement->thenbody);
     ControlFlow elsecf = ControlFlow::None;
     if (ifStatement->elsebody)
         elsecf = visit(elseScope, ifStatement->elsebody);
 
-    if (matches(thencf, ControlFlow::Returns | ControlFlow::Throws) && elsecf == ControlFlow::None)
+    if (matches(thencf, guardClauseFlows) && elsecf == ControlFlow::None)
```

Comment:
We can also say this. `ControlFlow::Breaks` and `ControlFlow::Continues` has never been used before, so this is equivalent anyhow regardless of the flagging.
```suggestion
    if (thencf != ControlFlow::None && elsecf == ControlFlow::None)
```